File: app/src/main/java/com/github/willjgriff/ethereumwallet/ui/createaccount/di/CreateAccountComponent.java
package com.github.willjgriff.ethereumwallet.ui.createaccount.di;

import com.github.wiljgriff.ethereumwallet.di.AppComponent;
import com.github.willjgriff.ethereumwallet.di.FunctionScope;
import com.github.willjgriff.ethereumwallet.ui.createaccount.CreateAccountController;

import dagger.Component;

/**
 * Created by Will on 03/02/2017.
 */

@Component(dependencies = AppComponent.class)
@FunctionScope
public interface CreateAccountComponent {

	void inject(CreateAccountController createAccountController);
}

File: app/src/main/kotlin/com/github/willjgriff/ethereumwallet/ui/screens/createaccount/di/CreateAccountComponent.kt
package com.github.willjgriff.ethereumwallet.ui.screens.createaccount.di

import com.github.willjgriff.ethereumwallet.di.AppComponent
import com.github.willjgriff.ethereumwallet.di.AppInjector
import com.github.willjgriff.ethereumwallet.di.ControllerScope
import com.github.willjgriff.ethereumwallet.ui.screens.createaccount.CreateAccountController
import dagger.Component

/**
 * Created by williamgriffiths on 03/05/2017.
 */
@Component(dependencies = arrayOf(AppComponent::class))
@ControllerScope
interface CreateAccountComponent {

    fun inject(createAccountController: CreateAccountController)
}

fun CreateAccountController.injectPresenter() {
    DaggerCreateAccountComponent
            .builder()
            .appComponent(AppInjector.appComponent)
            .build()
            .inject(this)
}